pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/pubsub/DSMtasksSubscription.java

```diff
@@ -92,11 +103,30 @@ public static void subscribeDSMtasks(String projectId, String subscriptionId) {
                 }
             }
         };
-        Subscriber subscriber = null;
+
         ProjectSubscriptionName resultSubName = ProjectSubscriptionName.of(projectId, subscriptionId);
         ExecutorProvider resultsSubExecProvider = InstantiatingExecutorProvider.newBuilder().setExecutorThreadCount(1).build();
-        subscriber = Subscriber.newBuilder(resultSubName, receiver).setParallelPullCount(1).setExecutorProvider(resultsSubExecProvider)
-                .setMaxAckExtensionPeriod(org.threeten.bp.Duration.ofSeconds(120)).build();
+
+        var subscriberBuilder  = Subscriber.newBuilder(resultSubName, receiver)
+                .setParallelPullCount(1)
+                .setExecutorProvider(resultsSubExecProvider)
+                .setMaxAckExtensionPeriod(org.threeten.bp.Duration.ofSeconds(120));
+        
+        if (!StringUtils.isBlank(pubSubHost)) {
+            var channel = ManagedChannelBuilder
+                .forTarget(pubSubHost)
+                .usePlaintext()
+                .build();
+            
+            subscriberBuilder.setChannelProvider(FixedTransportChannelProvider.create(GrpcTransportChannel.create(channel)));
+        }
+
+        if (emulator) {
+            subscriberBuilder.setCredentialsProvider(NoCredentialsProvider.create());
+        }
+
+        var subscriber = subscriberBuilder.build();
```

Comment:
This setup code for subscribers appears in multiple spots and is a prime candidate for a DRY refactoring, but I don't see a clear spot for the code to live. I'm open to suggestions.